sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeFixedWidthAggregationMap.java

```diff
@@ -55,9 +67,19 @@
   private final BytesToBytesMap map;
 
   /**
+   * A object pool for these in keys.
+   */
+  private final UniqueObjectPool keyPool;
+
+  /**
+   * A object pool for those in aggregation buffers.
+   */
+  private final ObjectPool bufferPool;
```

Comment:
I think `bufferPool` is a slightly confusing name, since I think that readers may confuse it with a memory page buffer pool used in databases.  Instead, maybe naming it something like "valueObjectPool" or "aggregationBufferObjectPool" would make this clearer.